### ISSUE TITLE
Using extensions to handle "soft enums"

### ISSUE THREAD

**Issue by JasonBenett** [DoclerLabs/api-client-generator#44]
Hello!

Currently, when the OpenApi spec defines an enum type, we are creating in the generated client a list of valid values. 

Imo, it would be interesting to be able to have an enum type in the spec (because this is the reality of the API) but, in the generated client, having a simple string instead. We could for example support openapi extensions like this:

```yaml
paths:
  '/v1/my/awesome/path':
    post:
      parameters:
        - schema:
             type: string
             enum: [value1, value2, value3]
             x-soft-enum: true
```

The main advantage is that when we add a new authorized value in the API, the client will not generate any error because it does not handles this value yet. If the value is not supported, the API is supposed to catch it. 

WDYT?

**Comment by vsouz4** [comment]
@JasonBenett I've experienced the same situation couple months ago, API started responding new enum value and the clients started failing. Client shouldn't start failing out of the blue because the API got updated, so that makes me think we should remove this validation altogether. We can add some extension later if needed in case we want to give the option for the client to have this validation, but by default I think we shouldn't have it. We can remove it and release a major version, wdyt?
